File: CPP/160_IntersectionOfTwoLinkedLists.cpp
// 160 - Intersection of Two Linked Lists

ListNode *getIntersectionNode(ListNode *headA, ListNode *headB) {
    unordered_map<ListNode*, int> hash;
    ListNode* curr = headA;

    while(curr != nullptr) {
        hash[curr] = curr->val;
        curr = curr->next;
    }

    curr = headB;

    while(curr != nullptr) {
        if(hash.find(curr) != hash.end()) {
            return curr;
        }
        curr = curr->next;
    }

    return nullptr;
}

File: Java/160_IntersectionOfTwoLinkedLists.java
// 160 - Intersection of Two Linked Lists

public ListNode getIntersectionNode(ListNode headA, ListNode headB) {
    Map<ListNode, Integer> hash = new HashMap<>();
    ListNode curr = headA;

    while(curr != null) {
        hash.put(curr, curr.val);
        curr = curr.next;
    }

    curr = headB;

    while(curr != null) {
        if(hash.containsKey(curr)) {
            return curr;
        }
        curr = curr.next;
    }

    return null;
}